esphome/components/mqtt/__init__.py

```diff
@@ -296,6 +297,7 @@ def validate_fingerprint(value):
                     cv.Optional(CONF_QOS, default=0): cv.mqtt_qos,
                 }
             ),
+            cv.Optional(CONF_PUBLISH_NAN_AS_NONE): cv.boolean,
```

Comment:
Default values should go in the schema
```suggestion
            cv.Optional(CONF_PUBLISH_NAN_AS_NONE, default=False): cv.boolean,
```